plexpy/http_handler.py

```diff
@@ -30,94 +33,144 @@ class HTTPHandler(object):
     Retrieve data from Plex Server
     """
 
-    def __init__(self, host, port, token, ssl_verify=True):
-        self.host = host
-        self.port = str(port)
+    def __init__(self, urls, token=None, timeout=10, ssl_verify=True):
+        if isinstance(urls, basestring):
+            self.urls = urls.split() or urls.split(',')
+        else:
+            self.urls = urls
+
         self.token = token
+        if self.token:
+            self.headers = {'X-Plex-Token': self.token}
+        else:
+            self.headers = {}
+
+        self.timeout = timeout
         self.ssl_verify = ssl_verify
 
-    """
-    Handle the HTTP requests.
+        self.valid_request_types = ('GET', 'POST', 'PUT', 'DELETE')
 
-    Output: object
-    """
     def make_request(self,
-                     uri=None, proto='HTTP',
-                     request_type='GET',
+                     uri=None,
                      headers=None,
+                     request_type='GET',
                      output_format='raw',
                      return_type=False,
                      no_token=False,
-                     timeout=None):
+                     timeout=None,
+                     callback=None):
+        """
+        Handle the HTTP requests.
+
+        Output: list
+        """
+
+        self.uri = uri
+        self.request_type = request_type.upper()
+        self.output_format = output_format.lower()
+        self.return_type = return_type
+        self.callback = callback
+        self.timeout = timeout or self.timeout
+
+        if self.request_type not in self.valid_request_types:
+            logger.debug(u"HTTP request made but unsupported request type given.")
+            return None
 
-        if timeout is None:
-            timeout = plexpy.CONFIG.PMS_TIMEOUT
+        if uri:
+            request_urls = [urljoin(url, self.uri) for url in self.urls]
 
-        valid_request_types = ['GET', 'POST', 'PUT', 'DELETE']
+            if no_token and headers:
+                self.headers = headers
+            elif headers:
+                self.headers.update(headers)
 
-        if request_type.upper() not in valid_request_types:
-            logger.debug(u"HTTP request made but unsupported request type given.")
+            responses = []
+            for r in self._http_requests_pool(request_urls):
```

Comment:
This is only temporary to make sure it works. I'm only working with one URL at a time at the moment.